<!-- Hello, fancy seeing you here, feel free to inspect my code, you will not find anything interesting. -->

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Gurpinder</title>
    <style>
        body {
            background-color: black;
            color: #33ff33;
            font-family: monospace;
            white-space: pre-line;
            margin: 20px;
            max-width: 80ch;
        }
        .terminal {
            padding: 20px;
        }
        h1, h2,h4, p {
        margin: 0;
        } 
        a {
            color: #33ff33;
            text-decoration: underline;
        }
        .blink::after {
            content: "_";
            display: inline-block;
            animation: blink 1s infinite;
        }
        @keyframes blink {
            50% { opacity: 0; }
        }
    </style>
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-M5YDEHJ8CF"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-M5YDEHJ8CF');
</script>
</head>
    


<body>
    <h1><div class="terminal" id="name"></div></h1>
        <script>
            const text = "Gurpinder S. Sidhu"
            
            let i = 0;
            function typeWriter() {
                if (i < text.length) {
                    document.getElementById("name").innerHTML += text.charAt(i);
                    i++;
                    setTimeout(typeWriter, 25);
                } else {
                    document.getElementById("name").innerHTML += '<span class="blink"></span>';
                }
            }
            
            window.onload = typeWriter;
    </script>
    <div class="terminal">
        <p>
            currently: postdoctoral scientist (computational biology), earlham institute, uk (oct. 2025 - present)
            
            last seen working as: 
            - phd student, john innes centre, uk (sep. 2021 - sep. 2025)
            - graphic design and web dev intern at iceland ocean cluster, iceland (mar. 2023 - may 2023) 
            - bioinformatician in steve penfield's lab, john innes centre, uk (jul. 2021 - sep. 2021) 
        </p>
            <h2>> about me </h2>
        <p>
            i am a computational biologist passionate about using mathematical and computational frameworks to solve biological problems. i can write decent quality code, can train a machine learning model or two and wrangle large datasets. i am always eager learn more about cool techniques and computers in general. i can also work in the lab, without destroying everything, if required. 
            
            when i am not doing science, i like to do graphic design and hike tall masses of land. 
        </p>
        <h2>> how to reach me?</h2>
        <p>
            email: <a href="mailto:gurpinder-singh.sidhu@jic.ac.uk">gurpinder-singh.sidhu@jic.ac.uk</a>
            github: <a href="https://github.com/Gurpinder98">github.com/Gurpinder98</a>
            blueSky: <a href="https://bsky.app/profile/grpndr.bsky.social">@grpndr.bsky.social</a>
            linkedin: <a href="https://www.linkedin.com/in/gurpinder-s-s-a75569268/">linkedin.com/in/gurpinder-s-s-a75569268/</a>
        </p>
        <h2>> what can i do?</h2>
        <p>
            technical skills: bioinformatics (experience with sequencing data, transcriptomic data, time-series data), data analysis, machine learning (experience with tensorflow, gpflow, keras), web development (experience with flask, html, css, javascript, wordpress), git, hpc (experience with slurm, snakemake)

            programming and scripting: python (★★★★★), r (★★★★★), bash (★★★★☆), c (★★★☆☆), rust (★★☆☆☆)

            lab skills: plant biology (tissue sampling), molecular biology (rna isolation), microscopy, data collection            
        </p>
        <h2>> select projects</h2>
        <p>
            - regulation of flowering time within <i>brassica napus</i>. (phd project). i am currently working on inferring flowering time gene regulatory networks in <i>brassica napus</i> using time-series gene expression data. as part of this project, i have implemented gene regulatory network inference methods (see <a href="https://github.com/Gurpinder98/Gaussian_network_inference">github</a>), a reciprocal blast utility for gene mappings (see <a href="https://github.com/Gurpinder98/boss">github</a>) and helped in testing of a new curve registration method (see <a href="https://ruthkr.github.io/greatR/">greatR</a>).

            - a bayesian framework for differential gene expression. i collaborated with a team of researchers to develop a bayesian framework for ranking genes based on their statistical evidence for differential expression. i contributed to the development of the framework, the testing of method with available r packages (see <a href="https://github.com/Morris-Research-Group/bayexpress">github</a>) and the writing of the manuscript.



        <h2>> select publications</h2>
        <p>
            - Hoerbst F, <strong>Sidhu GS</strong>, Tomkins M, Morris RJ. What is a differentially expressed gene? bioRxiv. 2025. doi: <a href="https://doi.org/10.1101/2025.01.31.635902">10.1101/2025.01.31.635902</a>
            
            - Hoerbst F, <strong>Sidhu GS</strong>, Omori T, Tomkins M, Morris RJ. A Bayesian framework for ranking genes based on their statistical evidence for differential expression. bioRxiv. 2025. doi: <a href = "https://doi.org/10.1101/2025.01.20.633909">10.1101/2025.01.20.633909</a>

            for the full list, see <a href="https://scholar.google.com/citations?user=YagxPJ4AAAAJ">google scholar</a>.
        </p>
    </div>
</body>
</html>
